src/main/resources/assets/js/components/Marathon.jsx

```diff
@@ -43,11 +49,61 @@ define([
         deploymentsFetchState: States.STATE_LOADING,
         fetchState: States.STATE_LOADING,
         modalClass: null,
+        route: null,
         tasksFetchState: States.STATE_LOADING
       };
     },
 
     componentDidMount: function() {
+      var router = this.props.router;
+
+      function callRoute(routeFunc) {
+        if(this.state.fetchState === States.STATE_LOADING) {
+          setTimeout(callRoute.bind(this, routeFunc), 300);
+        } else {
+          routeFunc();
+        }
+      };
+
+      var destroyModal = function() {
+        if(this.state.modalClass) {
+          this.modalDestroy();
+        }
+      }.bind(this);
+
+      // set route in state for all pages
+      router.on("route", function (route, params) {
+        var routeName = _.str.capitalize(route);
```

Comment:
Cool, yeah. Let's do that. I guess it is because the library checks for it, when it is included, but better be on the safe side :)